File: src/main/java/com/api/twitter/security/application/handler/SecurityExceptionHandler.java
package com.api.twitter.security.application.handler;

import com.api.twitter.common.dto.ApiResponse;
import com.api.twitter.security.application.exception.ExpiredTokenException;
import com.api.twitter.security.application.exception.InvalidTokenException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

@ControllerAdvice
public class SecurityExceptionHandler {
    @ExceptionHandler(ExpiredTokenException.class)
    public ResponseEntity<ApiResponse<Void>> expiredTokenHandler(ExpiredTokenException exception){
        return ResponseEntity
                .status(HttpStatus.BAD_REQUEST)
                .body(ApiResponse.fail(exception.getMessage()));
    }

    @ExceptionHandler(InvalidTokenException.class)
    public ResponseEntity<ApiResponse<Void>> invalidTokenHandler(InvalidTokenException exception){
        return ResponseEntity
                .status(HttpStatus.BAD_REQUEST)
                .body(ApiResponse.fail(exception.getMessage()));
    }
}

File: src/main/java/com/api/twitter/security/application/handler/SecurityExceptionHandler.kt
package com.api.twitter.security.application.handler

import com.api.twitter.common.dto.ApiResponseDTO
import com.api.twitter.security.application.exception.ExpiredTokenException
import com.api.twitter.security.application.exception.InvalidTokenException
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler

@ControllerAdvice
class SecurityExceptionHandler {
    @ExceptionHandler(ExpiredTokenException::class)
    fun expiredTokenHandler(exception: ExpiredTokenException): ResponseEntity<ApiResponseDTO<Void>> {
        return ResponseEntity
            .status(HttpStatus.BAD_REQUEST)
            .body(ApiResponseDTO.fail(exception.message))
    }

    @ExceptionHandler(InvalidTokenException::class)
    fun invalidTokenHandler(exception: InvalidTokenException): ResponseEntity<ApiResponseDTO<Void>> {
        return ResponseEntity
            .status(HttpStatus.BAD_REQUEST)
            .body(ApiResponseDTO.fail(exception.message))
    }
}
